src/Compilers/Core/Portable/SourceGeneration/Nodes/NodeStateTable.cs

```diff
@@ -199,7 +203,7 @@ public bool TryRemoveEntries(TimeSpan elapsedTime, ImmutableArray<(IncrementalGe
                 }
 
                 // Mark the corresponding entries to this node execution in the previous table as removed.
-                var previousEntries = _previous._states[_states.Count].AsRemoved();
+                var previousEntries = _previous._states[_states.Count].AsRemovedDueToInputRemoval();
```

Comment:
nit: this is the only place where we use `AsRemovedDueToInputRemoval()`. It may be helpful to adjust the comment for more details on why.